### PR TITLE
Azure and Helm deployment improvements - usability and security

### PR DESCRIPTION
**Pull request by kaizimmerm**
Introduced the following mechanisms to improve security and usability for Azure and Helm deployments:

Azure:
* Deployment into dedicated virtual network.
* [CosmosDB Vnet service endpoint](https://azure.microsoft.com/de-de/blog/virtual-network-service-endpoints-for-azure-cosmos-db-is-now-generally-available/) ensures that only nodes inside chosen network (i.e. the AKS cluster) can access the Database.
* Static IP for Ditto in Azure Loadbalancer including DNS name.

Helm:
* Introduced [k8s network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) to restrict inner cluster communication.
* Introduced more variables and fixed a few (MongoDb SSL configs)

### CODE REVIEW

File: deployment/azure/arm/dittoInfrastructureDeployment.json [hunk 1]
```diff
@@ -44,34 +92,59 @@
...
             "type": "Microsoft.Resources/deployments",
             "apiVersion": "2018-05-01",
+            "dependsOn": [
+                "[concat('Microsoft.Resources/deployments/', 'dittoVnetDeployment')]"
+            ],
             "properties": {
                 "mode": "Incremental",
                 "templateLink": {
-                    "uri": "https://raw.githubusercontent.com/eclipse/ditto/master/deployment/azure/arm/templates/kubernetesDeploy.json"
+                    "uri": "https://raw.githubusercontent.com/kaizimmerm/ditto/aksvnet/deployment/azure/arm/templates/kubernetesDeploy.json"
```
Comment (thjaeckle) [inline deployment/azure/arm/dittoInfrastructureDeployment.json]: Is it intended to use your `kaizimmerm/ditto` fork here?

File: deployment/azure/arm/dittoInfrastructureDeployment.json [hunk 2]
```diff
@@ -81,14 +154,48 @@
...
             "type": "Microsoft.Resources/deployments",
             "apiVersion": "2018-05-01",
+            "dependsOn": [
+                "[concat('Microsoft.Resources/deployments/', 'dittoVnetDeployment')]"
+            ],
             "properties": {
                 "mode": "Incremental",
                 "templateLink": {
-                    "uri": "https://raw.githubusercontent.com/eclipse/ditto/master/deployment/azure/arm/templates/cosmosDBDeploy.json"
+                    "uri": "https://raw.githubusercontent.com/kaizimmerm/ditto/aksvnet/deployment/azure/arm/templates/cosmosDBDeploy.json"
```
Comment (thjaeckle) [inline deployment/azure/arm/dittoInfrastructureDeployment.json]: Same here .. `kaizimmerm/ditto`

File: deployment/azure/arm/dittoInfrastructureDeployment.json (hunk 1 above)
Comment (kaizimmerm) [reply deployment/azure/arm/dittoInfrastructureDeployment.json]: Yes and no, usually @ghandim wants to test this but it won't work until the PR is merged with the eclipse repo URI. I will create a second commit later that changes this to the final URI.

**Comment by thjaeckle** [comment]
@ghandim could you please try that again on Azure?

**Comment by ghandim** [comment]
I will give it  a try.

**Comment by ghandim** [comment]
@kaizimmerm @thjaeckle LGTM :+1:

**Comment by kaizimmerm** [comment]
I will update the URLs then and merge with master again.

**Comment by ghandim** [comment]
@kaizimmerm One more thing. Could you please also update the ditto release version to latest and greatest 0.9.0? Tested the setup yesterday with this version.

**Pull request merged**